generator/osm2meta.hpp

```diff
@@ -29,7 +29,7 @@ struct MetadataTagProcessorImpl
   string ValidateAndFormat_flats(string const & v) const;
   string ValidateAndFormat_internet(string v) const;
   string ValidateAndFormat_height(string const & v) const;
-  string ValidateAndFormat_building_levels(string const & v) const;
+  string ValidateAndFormat_building_levels(string v) const;
```

Comment:
void f(string s); создает копию параметра в теле функции, которую можно менять в теле, но наружу она не выползет.
void f(string & s); изменения параметра в теле автоматом меняют внешнюю переменную, по ссылке.
void f(string const & s); ссылка константная, подразумевается, что внутри метода не нужно менять переменную. Но если нужно, придется вручную создавать копию, типа string s2 = s; чтобы избежать “ручного” создания копии, используется передача по значению.


> On Mar 31, 2016, at 20:19, Ilya Zverev <notifications@github.com> wrote:
> 
> In generator/osm2meta.hpp <https://github.com/mapsme/omim/pull/2670#discussion_r58092254>:
> 
> > @@ -29,7 +29,7 @@ struct MetadataTagProcessorImpl
> >    string ValidateAndFormat_flats(string const & v) const;
> >    string ValidateAndFormat_internet(string v) const;
> >    string ValidateAndFormat_height(string const & v) const;
> > -  string ValidateAndFormat_building_levels(string const & v) const;
> > +  string ValidateAndFormat_building_levels(string v) const;
> А почему некоторые функции изменяют исходную строку? Зачем, если она потом не используется?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2670/files/7ef4557f7e632554725d907e55c5b32a6566ea47#r58092254>